### PR TITLE
[Moe] Fix flashinfer_trtllm silently dropping swiglu_limit clamped SwiGLU activation

### LINKED ISSUE #39797: [Question] GLM-5.3-Flash on 4x GB200: `--moe-runner-backend flashinfer_trtllm` scores ~1.3 gsm8k points below `deep_gemm` in repeated runs (6 vs 3 runs); one `triton` run agrees with deep_gemm
### Describe the bug

In the TEP4 configuration we tested (TP4 + EP4, trtllm DSA backends, fp8 KV cache, `--moe-runner-backend flashinfer_trtllm` — the cookbook's GB200 cell is TP4 without EP and gave the same score, see below), the mean of six GLM-5.3-Flash 5-shot greedy gsm8k runs is ~1.3 points below the mean of three runs of the same server with `--moe-runner-backend deep_gemm`, and every one of the six trtllm-runner scores is below every one of the three deep_gemm scores (closest pair 0.9272 vs 0.9280); it is also below vLLM on the same hardware. A single `triton` run landed with deep_gemm. lm-evaluation-harness 0.4.13, `local-completions`, one prompt per request, all 1319 questions, flexible-extract.

Same server command repeated (one server start per run, 32 concurrent requests; sample sd):

| configuration | runs | mean | sample sd | values |
|---|---|---|---|---|
| SGLang TEP4 `--moe-runner-backend flashinfer_trtllm` | 6 | 0.9188 | 0.0048 | 0.9174, 0.9143, 0.9143, 0.9189, 0.9272, 0.9204 |
| SGLang TEP4 `--moe-runner-backend deep_gemm` | 3 | 0.9315 | 0.0032 | 0.9325, 0.9280, 0.9340 |
| SGLang TEP4 `--moe-runner-backend triton` | 1 | 0.9303 | | |
| vLLM TP4, FlashInfer trtllm FP8 MoE backend (same kernel) | 5 | 0.9327 | 0.0031 | 0.9340, 0.9272, 0.9348, 0.9340, 0.9333 |

Single trtllm-runner runs under changed conditions stay in the same range: TP4 without EP 0.9158; `--kv-cache-dtype bfloat16` 0.9151; strictly sequential requests 0.9174 / 0.9174; the exact cookbook cell on the cookbook image (`lmsysorg/sglang:glm-5.3-flash`, `0.0.0.dev1+gf609d677b`) 0.9181; a patch casting routing logits/bias to fp32 before the kernel (they already are fp32) 0.9265 / 0.9174. Not controls: "auto" DSA backends resolve to trtllm anyway (0.9181), and `--disable-shared-experts-fusion` is already set automatically for the trtllm runner (0.9212). Paired exact McNemar tests of the three original trtllm runs vs the first deep_gemm run: p = 0.020 / 0.0049 / 0.0043. The discordant questions are "different numeric answer", not extraction failures.

### What we checked (all negative)

- Same FlashInfer 0.6.18 `trtllm_fp8_block_scale_moe` kernel as vLLM's backend. Logged at run time (prefill and decode calls): fp32 router logits, fp32 correction bias, fp8 activations with a contiguous fp32 [H/128, T] scale tensor, n_group 1 / topk_group 1 / top_k 8 / routed_scaling_factor 2.5 / DeepSeekV3 routing, 72 local of 288 experts, `use_shuffled_weight=False` (`WeightLayout.MajorK`).
- Kernel-level check on real layer weights (layers 10 and 20; FlashInfer's own tests/moe reference helpers): with fp32 routing inputs the kernel is within ~4.5% relative L2 of an fp32 evaluation of the same FP8 tensors; unshuffled vs shuffled layouts bitwise identical; same for 1–4096 tokens, autotune on/off, in-kernel vs pre-routed top-k. Feeding the correction bias in bf16 would raise the error to 54% (the bias is 6.18–6.56 with a 0.38 spread), but SGLang passes fp32.
- Dumps of the exact tensors the runner hands to the kernel (start-up forwards): weights and block scales bitwise equal to the checkpoint (kernel [up; gate] row order), activation quantisation correct, fp32 logits and bias on prefill- and decode-sized calls. On the first two real lm-eval prompts (911 / 952 tokens, the same prompts for both runners; prefill only — decode runs as CUDA-graph replay and cannot be hooked from Python) the kernel output vs an fp32 evaluation of the checkpoint experts on each runner's own layer inputs: trtllm median 5.0% relative L2 (max 5.6%), deep_gemm median 5.3% (max 6.1%; it runs on UE8M0-requantised weights). Same reference definition, same prompts, but each runner sees its own layer inputs and routing, so not a paired comparison; the local per-layer errors are of the same order and do not explain the end-to-end gap.

### Question

Switching the runner reproducibly moves the score by ~1.3 points; we have not isolated the mechanism. What else differs between the `flashinfer_trtllm` and `deep_gemm` MoE paths for this model that could cost ~1.3 gsm8k points end to end? Things we could not test from outside: decode steps under CUDA-graph replay (padding / `num_token_non_padded` handling in the in-kernel routing), and anything that only shows on real-text activations.

### Reproduction

sglang `0.0.0.dev1+g00143e9c2` (`lmsysorg/sglang:dev` pulled 2026-09-11 01:04 UTC), FlashInfer 0.6.18, 4x GB200 (one NVL72 tray), `zai-org/GLM-5.3-Flash` revision `eb9eb208eb0d988989d07a6a12d0fdeb5f52574a`, lm-evaluation-harness 0.4.13:
```
python -m sglang.launch_server --model-path zai-org/GLM-5.3-Flash --revision eb9eb208eb0d988989d07a6a12d0fdeb5f52574a \
  --served-model-name model --host 127.0.0.1 --port 8000 \
  --tp-size 4 --ep-size 4 --dsa-prefill-backend trtllm --dsa-decode-backend trtllm --kv-cache-dtype fp8_e4m3 \
  --moe-runner-backend flashinfer_trtllm --context-length 16384 --disable-radix-cache
lm_eval --model local-completions \
  --model_args model=model,base_url=http://127.0.0.1:8000/v1/completions,num_concurrent=32,max_retries=3,tokenized_requests=False,tokenizer_backend=None \
  --tasks gsm8k --num_fewshot 5 --batch_size 1 --gen_kwargs temperature=0,max_gen_toks=512 --seed 1234 --log_samples --output_path out
```
Replace `--moe-runner-backend flashinfer_trtllm` with `deep_gemm` for the comparison run. Scores above are `exact_match,flexible-extract`. Same-tray throughput cost of deep_gemm on 8k-in/1k-out random prompts: 0.87× (c=1) to 0.98× (c=128) of the trtllm runner.

### PR DESCRIPTION
**Pull request by Dovis01**
## Motivation

Models with a clamped SwiGLU activation (e.g. GLM-5.3-Flash / GLM-5-Next, Qwen3-Next style) plumb the clamp limit through `FusedMoE(swiglu_limit=...)` -> `MoeRunnerConfig.swiglu_limit`. All MoE runner backends are supposed to honor it: `triton` passes it into the fused kernel, `deep_gemm` applies it via `silu_and_mul_clamp` / `_apply_swiglu_limit`, but `flashinfer_trtllm` **silently ignores it**.

The flashinfer_trtllm FP8 path only materializes `gemm1_clamp_limit` (the TRT-LLM kernel parameter), which stays `None` for models that use the `swiglu_limit` field. With no error and no warning, the expert activation degrades from the trained `silu(clamp(gate, max=L)) * clamp(up, -L, L)` (with L = 10.0) to a plain `silu(gate) * up`, which measurably hurts accuracy.

On GLM-5.3-Flash FP8 (blockwise 128x128 FP8, 288 routed experts, top-8, 1 shared expert, `swiglu_limit=10.0`), GSM8K 5-shot with only `--moe-runner-backend` varying:

| moe_runner_backend | GSM8K strict-match (runs) | mean |
|---|---|---|
| triton | 0.9356 / 0.9363 | 0.9360 |
| deep_gemm | 0.9249 / 0.9287 / 0.9295 / 0.9310 / 0.9340 / 0.9348 | 0.9305 |
| flashinfer_trtllm (before) | 0.9265 / 0.9280 / 0.9287 | 0.9277 |
| **flashinfer_trtllm (this fix)** | **0.9348 / 0.9371** | **0.9360** |

Before/after ranges do not overlap (within-group variance is ~0.2pp); after the fix flashinfer_trtllm matches the triton baseline exactly. fix https://github.com/sgl-project/sglang/issues/39797. Thx for finding this issue. @NolenLiang

## Modifications

Single-file change in `python/sglang/srt/layers/quantization/fp8.py`, `Fp8MoEMethod._prepare_flashinfer_trtllm_activation_params()`: when `gemm1_clamp_limit` is not explicitly set, fall back to `swiglu_limit`.

Why here and not in the model code: this function is the only consumer that builds the TRT-LLM SwiGLU parameters (`gemm1_alpha/beta/clamp_limit`), so the two naming pipelines (`gemm1_clamp_limit` used by gpt_oss / bailing_moe_v3, `swiglu_limit` used by DeepseekV2MoE-family models) converge at a single point; `triton` consumes both fields independently (`gemm1_limit` and `swiglu_limit`), so setting both at the model level would change behavior of unrelated backends, while this fallback is scoped to the flashinfer_trtllm preparation only. Models that already set `gemm1_clamp_limit` explicitly (gpt_oss, bailing_moe_v3) are unaffected, and the same silent drop is fixed for every other model that routes the clamp through `swiglu_limit` (e.g. Qwen3-Next) whenever flashinfer_trtllm is used.

The TRT-LLM kernel natively supports the clamp: with `ActivationType.Swiglu`, `gemm1_clamp_limit=L` computes `X2 = clamp(X2, max=L)`, `X1 = clamp(X1, -L, L)`, `out = X2 * sigmoid(alpha * X2) * (X1 + beta)` (alpha defaults to 1.0, beta to 0.0), which matches the reference `swiglu_clamped` in `models/glm5_next.py` exactly. Verified against flashinfer 0.6.18 (`trtllm_fp8_block_scale_moe`, `Fp8QuantizationType.DeepSeekFp8`).

## Accuracy Tests

Model: `zai-org/GLM-5.3-Flash` (FP8, blockwise 128x128), 4x B200, TP=4/EP=4. Launch (only `--moe-runner-backend` varies between runs):

```bash
 python3 -m sglang.launch_server \
  --model-path zai-org/GLM-5.3-Flash \
  --tp-size 4 --ep-size 4 --mem-fraction-static 0.75 \
  --dsa-prefill-backend trtllm --dsa-decode-backend trtllm \
  --kv-cache-dtype fp8_e4m3 \
  --moe-runner-backend flashinfer_trtllm \
  --context-length 16384 --disable-radix-cache \
  --host 0.0.0.0 --port 8000 --served-model-name glm-5-next-fp8
```

Eval (lm-evaluation-harness, `local-completions`):

```bash
lm_eval --model local-completions \
  --model_args "model=glm-5-next-fp8,base_url=http://127.0.0.1:8000/v1/completions,num_concurrent=32,max_retries=3,tokenized_requests=False,tokenizer_backend=None" \
  --tasks gsm8k --num_fewshot 5 --batch_size 1 \
  --gen_kwargs temperature=0,max_gen_toks=512 --seed 1234 --log_samples
```

Results (exact_match, strict-match):

| config | run 1 | run 2 | run 3 | mean |
|---|---|---|---|---|
| flashinfer_trtllm before | 0.9287 | 0.9265 | 0.9280 | 0.9277 |
| flashinfer_trtllm **after** | 0.9371 | 0.9348 | – | **0.9360** |
| triton (baseline) | 0.9363 | 0.9356 | – | 0.9360 |

Fixed server sanity-checked with deterministic generation (temperature=0) on GSM8K-style prompts before running the evals.

## Speed Tests and Profiling

Not affected. The change only materializes one extra per-expert fp32 tensor (`[num_local_experts]`, 288 bytes here) once at weight-load time; the hot kernel call is unchanged (the `gemm1_clamp_limit` argument was already being passed, as `None`). No throughput regression expected or observed.

---
### CI States

Latest PR Test (Base): :hourglass_flowing_sand: [Run #35192393525](https://github.com/sgl-project/sglang/actions/runs/35192393525)
Latest PR Test (Extra): :x: [Run #35192393195](https://github.com/sgl-project/sglang/actions/runs/35192393195)
Latest PR Test (AMD ROCm 10): :hourglass_flowing_sand: [Run #35192393447](https://github.com/sgl-project/sglang/actions/runs/35192393447)

### CODE REVIEW

**Comment by hnyls2002** [comment]
/rerun-test test_fp8_moe_runner_ownership.py test_moe_runner_extensions.py
